### PR TITLE
Fix agent event and use direct output from agent

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: src/ragapp/backend/agents/single.py [hunk 1]
```diff
@@ -229,6 +241,7 @@ async def handle_tool_calls(self, ctx: Context, ev: ToolCallEvent) -> InputEvent
                     )
                 )
             except Exception as e:
+                print(e)
```
Comment (marcusschiesser) [inline src/ragapp/backend/agents/single.py:244]: not needed

File: src/ragapp/backend/routers/chat/vercel_response.py [hunk 2]
```diff
@@ -160,8 +161,12 @@ async def _chat_response_generator():
 
             if isinstance(result, AsyncGenerator):
                 async for token in result:
-                    final_response += token.delta
-                    yield self.convert_text(token.delta)
+                    if isinstance(token, AgentRunEvent):
+                        final_response += token.msg
+                        yield self.convert_text(token.msg)
+                    elif isinstance(token, ChatResponse):
```
Comment (marcusschiesser) [inline src/ragapp/backend/routers/chat/vercel_response.py:167]: why is this change needed?

File: src/ragapp/backend/agents/single.py [hunk 3]
```diff
@@ -215,6 +220,13 @@ async def handle_tool_calls(self, ctx: Context, ev: ToolCallEvent) -> InputEvent
                 continue
 
             try:
+                if ctx.data["streaming"] and ctx.data["return_tool_output"]:
+                    # Complete the workflow using the result from the last agent
```
Comment (marcusschiesser) [inline src/ragapp/backend/agents/single.py:224]:
```suggestion
                    # Complete the workflow using the result from the last tool
```

File: src/ragapp/backend/agents/single.py [hunk 4]
```diff
@@ -215,6 +220,13 @@ async def handle_tool_calls(self, ctx: Context, ev: ToolCallEvent) -> InputEvent
...
+                    result = await tool.astream_response(
```
[the remark this answers is not in the document]
Comment (leehuwuj) [reply src/ragapp/backend/agents/single.py:225]: Yes, but i don't think it's problem right now, since return_tool_output only applies to executors who aren't using any normal tool.

**Review by marcusschiesser — APPROVED** [review]
